### PR TITLE
refactor(api): add blow_out, pick_up_tip, drop_tip

### LINKED ISSUE #2483: Robot Singleton Refactor: Add misc pipette actions to hardware_control
Add blow_out, pick_up_tip, drop_tip functionality to HC
Modify `move` methods to accept speed input

### PR DESCRIPTION
**Pull request by sanni-t**
Closes #2483

## changelog

- added `blow_out(mount)`, `pick_up_tip(mount, presses, increment)`, drop_tip(mount)`
- changed all `move` methods to accept move speed. This enables us to implement `touch_tip` in protocol_api using only hardware_control `move`s
- added pick_up_tip test

## review requests

Test the three functions on a robot. 
Example test:
```
>>> import opentrons.hardware_control as hc
>>> import asyncio
>>> import opentrons.types as t
>>> l=asyncio.get_event_loop()
>>> a=hc.API.build_hardware_controller()
>>> l.run_until_complete(a.home())
>>> l.run_until_complete(a.cache_instruments())
>>> l.run_until_complete(a.move_to(t.Mount.LEFT,t.Point(11.6,75,63)))  # rough loc of first tip of P300 tiprack in slot 1
>>> l.run_until_complete(a.pick_up_tip(t.Mount.LEFT))
>>> l.run_until_complete(a.aspirate(t.Mount.LEFT))
>>> l.run_until_complete(a.dispense(t.Mount.LEFT))
>>> l.run_until_complete(a.blow_out(t.Mount.LEFT))
>>> l.run_until_complete(a.move_to(t.Mount.LEFT,t.Point(11.6,75,50)))  #move close to tiprack
>>> l.run_until_complete(a.drop_tip(t.Mount.LEFT))
```

### CODE REVIEW

File: api/src/opentrons/hardware_control/__init__.py [hunk 1]
```diff
@@ -300,7 +303,8 @@ def current_position(self, mount: top_types.Mount) -> Dict[Axis, float]:
 
     @_log_call
     async def move_to(
-            self, mount: top_types.Mount, abs_position: top_types.Point):
+            self, mount: top_types.Mount, abs_position: top_types.Point,
+            speed: float = None):
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/__init__.py]: Can we add info on the comment for what "speed" means if there's more than one axis in the move?

File: api/src/opentrons/hardware_control/__init__.py [hunk 2]
```diff
@@ -529,68 +549,145 @@ def config(self) -> robot_configs.robot_config:
...
-    async def air_gap(self, mount, volume=None):
-        pass
+        self._backend.set_active_current(Axis.of_plunger(mount),
+                                         this_pipette.config.plunger_current)
+        try:
+            await self._move_plunger(
+                mount, this_pipette.config.plunger_positions['blow_out'])
+        except Exception:
+            self._log.exception('Blow out failed')
+            raise
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/__init__.py]: Set `current_volume = 0` in a `finally` clause here?

File: api/src/opentrons/hardware_control/__init__.py [hunk 3]
```diff
@@ -529,68 +549,145 @@ def config(self) -> robot_configs.robot_config:
...
+    async def _shake_off_tips(self, mount):
+        # tips don't always fall off, especially if resting against
+        # tiprack or other tips below it. To ensure the tip has fallen
+        # first, shake the pipette to dislodge partially-sealed tips,
+        # then second, raise the pipette so loosened tips have room to fall
+        shake_off_dist = SHAKE_OFF_TIPS_DISTANCE
+        # TODO: ensure the distance is not >25% the diameter of placeable
+        shake_pos = top_types.Point(0, 0, -shake_off_dist)  # move left
+        await self.move_rel(mount, shake_pos, speed=SHAKE_OFF_TIPS_SPEED)
+        shake_pos = top_types.Point(0, 0, 2*shake_off_dist)    # move right
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/__init__.py]: The comments here say "move left" and "move right" but these points seem to indicate a move in +-z  should these be `Point(-shake_off_dist, 0, 0)` and `Point(shake_off_dist, 0, 0)`?

Comment (sanni-t) [reply api/src/opentrons/hardware_control/__init__.py]: woops! You're right

File: api/src/opentrons/hardware_control/__init__.py (hunk 2 above)
Comment (sanni-t) [reply api/src/opentrons/hardware_control/__init__.py]: Is it okay to set current_volume = 0 even if blow_out fails?

File: api/src/opentrons/hardware_control/__init__.py (hunk 1 above)
Comment (sanni-t) [reply api/src/opentrons/hardware_control/__init__.py]: Totally

**Comment by sfoster1** [comment]
Tests OK on wolves. One note - you should actually move to more like `(11.6,72,50)` for the drop tip since the critical point is now the end of the tip.

**Comment by sfoster1** [comment]
Turns out the answer to the current thing is that the smoothie driver dwells pipette axes and dwell uses a lower current setting.

**Comment by sanni-t** [comment]
Added docstring explaining the situation with `active_current`

**Pull request merged**